cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -638,41 +642,40 @@ public static ClientBuilder bluemix() {
      * service instance.
      * <P>
      * Note: Specifying a service name is only required when multiple Cloudant services are
-     * available. If there is only a single bound Cloudant service instance then {@link #bluemix()}
-     * can also be used.
+     * available. If there is only a single bound Cloudant service instance then
+     * {@link #bluemix(String)} can also be used.
      * </P>
      *
-     * @param serviceName name of Bluemix service instance or {@code null} to try to use the only
+     * @param vcapServices contents of VCAP_SERVICES environment variable
+     * @param instanceName name of Bluemix service instance or {@code null} to try to use the only
      *                    available Cloudant service
      * @return a new ClientBuilder for the account
      *
      * @throws IllegalArgumentException if the Cloudant service instance is not available
-     * @throws IllegalStateException    if the VCAP_SERVICES environment variable is invalid or missing
      */
-    public static ClientBuilder bluemix(String serviceName) {
+    public static ClientBuilder bluemix(String vcapServices, String instanceName) {
         Gson gson = new GsonBuilder().create();
 
-        String vcap_env = System.getenv("VCAP_SERVICES");
-        if (vcap_env == null) {
-            throw new IllegalStateException("VCAP_SERVICES is not present.");
+        if (vcapServices == null) {
+            throw new IllegalArgumentException("VCAP_SERVICES is not present.");
```

Comment:
Since this is not about the environment variable being present any longer we should re-word this. How about:
VCAP_SERVICES JSON content was null.
or
VCAP_SERVICES JSON content must not be null.

Although I'm inclined to use the parameter name `vcapServices`, rather than the capitalized environment variable name since there is no guarantee it was from the env var in this new approach.

The [bluemix documentation](https://console.ng.bluemix.net/docs/cli/vcapsvc.html) says:
>The VCAP_SERVICES environment variable is a JSON object that contains information that you can use to interact with a service instance in Bluemix.

So maybe we should refer to the thing as vcapServices JSON information?
`The vcapServices JSON information was null.`